converter/biosamples.py

```diff
@@ -24,14 +24,15 @@ def __init__(self, default_domain=None):
     def convert(self, biomaterial: dict, additional_attributes: dict = None) -> Sample:
         domain = additional_attributes.get('domain')
         release_date = additional_attributes.get('release_date')
-        accession = additional_attributes.get('accession')
+        existing_accession = additional_attributes.get('accession')
 
         if not domain and not self.default_domain:
             raise MissingBioSamplesDomain()
         biomaterial_content = biomaterial.get('content', {})
         biomaterial_core = biomaterial_content.get('biomaterial_core', {})
+        sample_accession = self.__define_accession(existing_accession, biomaterial_core)
         sample = Sample(
-            accession=accession if accession else biomaterial_core.get('biosamples_accession'),
+            accession=sample_accession,
```

Comment:
there is a possibility of this being `None`. Is this permitted?

Reply:
@prabh-t Yes, it is fine. If it is `None` then BioStudies will create a new biosample